client/public/clientJS/profile.js

```diff
@@ -58,6 +58,65 @@ $(document).ready(() => {
 // About me edit. Click handler
 window.onload = function () {
   let buttonSaveProfile = document.getElementById('save-about-me-button');
+  let agencyAddressCheckBox = document.getElementById('agencyAddressCheckBox');
+
+  buttonSaveProfile.addEventListener('click', handleSaveAboutMeButtonClick);
+  agencyAddressCheckBox.addEventListener('click', handleCheckBoxClick);
+
+  function handleCheckBoxClick() {
+    if (agencyAddressCheckBox.checked) {
+      // if the address is already filled with data ask user to clear it first
+      // so that we dont accidentally overwrite it
+      if (checkIfAgencyAddressFormIsEmpty()) {
+        fetch('/users/agency/address')
+        .then((response) => response.json())
+        .then((data) => {
+          const agency = data.data;
+          if (agency.agencyAddress) {
+            updateAgencyAddressForm(agency);
+          }
+          else {
+            showToast("Agency address not present. Most likely that information is missing, contact administrator");
+          }
+        })
+        .catch((error) => {
+          showToast("Could not get agency address");
+        })
+      }
+      else {
+        showToast("Address form must be empty if we want to use agency address so that there is no accidental overwrite");
```

Comment:
the toast might not be shown long enough for errors like this I think. Is there a very to just ask the user to confirm if they want to overwrite the data?